python/prima/_nonlinear_constraints.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+        
+def process_single_nl_constraint(nlc):
+    '''
+    The Python interfaces receives the constraints as lb <= constraint(x) <= ub, 
+    but the Fortran backend expects the nonlinear constraints to be constr(x) <= 0.
+    Thus a conversion is needed.
+    
+    This function will take a NonlinearConstraint and return a new function which incorporates
+    the lower and upper bounds in such a way as to satisfy the expectation of the Fortran
+    backend. This new function first calls the original function from the provided NonlinearConstraint
+    and obtains the values, and then passes those values to a second function which combines
+    them with the lower and upper bounds as appropriate. The main work here is creating that
+    second function.
+    
+    In general to convert lb <= constraint(x) <= ub to newconstraint(x) <= 0, all we need to do
+    is define newconstraint(x) as [constraint(x) - ub, lb - constraint(x)]. There are some details
+    regarding the type and content of lb and ub as detailed below.
+    
+    
+    The upper and lower bounds could be either scalars or vectors, and so we have 4 possible
```

Comment:
I  do not think there are so many cases. 

We should first "broadcast" `lb` to a vector if it is given as a scalar, and do the same to `ub`. 

Then the pseudo code for the constraint conversion would be the following. It is branchless. 

```
non_trivial_lb = (lb > -inf)
non_trivial_ub = (ib < inf)
constr_lb = lb - constr
constr_ub = constr - ub 
constr = [constr_lb(non_trivial_lb) constr_ub(non_trivial_ub)]
```

Reply:
The issue with this approach was that to broadcast to a vector it is necessary to call the constraint function and I was trying to avoid doing so until later, so that I would not have to manage `nlconstr0` coming from different constraint functions with various lb and ub. However I ended up doing exactly that for reasons specified in https://github.com/libprima/prima/pull/124#issuecomment-2003173103.